### PR TITLE
Protect lba Get and Set with lock

### PR DESCRIPTION
**Pull request by iwanbk**
Part of #239

### CODE REVIEW

**Comment by iwanbk** [comment]
Fix below test.
In other words, it fixes deduped storage data race
```
$ go test -v -race -run ^TestDeduped
=== RUN   TestDedupedContent
INFO[06-05|13:33:38] embedded LedisDB Server ready and listening at 127.0.0.1:43077 module=global caller=redis.go:47
--- PASS: TestDedupedContent (0.01s)
=== RUN   TestDedupedContentForceFlush
INFO[06-05|13:33:38] embedded LedisDB Server ready and listening at 127.0.0.1:36935 module=global caller=redis.go:47
--- PASS: TestDedupedContentForceFlush (0.01s)
=== RUN   TestDedupedDeadlock
INFO[06-05|13:33:38] embedded LedisDB Server ready and listening at 127.0.0.1:39167 module=global caller=redis.go:47
==================
WARNING: DATA RACE
Write at 0x00c420930938 by goroutine 244:
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:88 +0x187
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:54 +0x1e2
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func1()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:450 +0xbd

Previous write at 0x00c420930938 by goroutine 235:
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:88 +0x187
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:54 +0x1e2
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func1()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:450 +0xbd

Goroutine 244 (running) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:466 +0x2c6
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107

Goroutine 235 (running) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:466 +0x2c6
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107
==================
==================
WARNING: DATA RACE
Write at 0x00c420ac10f8 by goroutine 39:
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:88 +0x187
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:54 +0x1e2
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Merge()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:82 +0x1b3
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func2()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:497 +0x37e

Previous write at 0x00c420ac10f8 by goroutine 386:
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:88 +0x187
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:54 +0x1e2
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Merge()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:82 +0x1b3
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func2()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:497 +0x37e

Goroutine 39 (running) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:501 +0x61b
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107

Goroutine 386 (running) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:501 +0x61b
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107
==================
==================
WARNING: DATA RACE
Write at 0x00c420ac10f8 by goroutine 94:
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:88 +0x187
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:54 +0x1e2
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Merge()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:82 +0x1b3
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func2()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:497 +0x37e

Previous write at 0x00c420ac10f8 by goroutine 386:
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:88 +0x187
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:54 +0x1e2
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Merge()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:82 +0x1b3
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func2()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:497 +0x37e

Goroutine 94 (running) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:501 +0x61b
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107

Goroutine 386 (finished) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:501 +0x61b
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107
==================
==================
WARNING: DATA RACE
Read at 0x00c42081b160 by goroutine 247:
  runtime.slicecopy()
      /home/ibk/go/src/runtime/slice.go:160 +0x0
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Get()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:118 +0x1d8
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Merge()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:59 +0x76
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func2()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:497 +0x37e

Previous write at 0x00c42081b160 by goroutine 403:
  [failed to restore the stack]

Goroutine 247 (running) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:501 +0x61b
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107

Goroutine 403 (finished) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:501 +0x61b
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107
==================
==================
WARNING: DATA RACE
Read at 0x00c42081b060 by goroutine 217:
  runtime.slicecopy()
      /home/ibk/go/src/runtime/slice.go:160 +0x0
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Get()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:118 +0x1d8
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Merge()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:59 +0x76
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func2()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:497 +0x37e

Previous write at 0x00c42081b060 by goroutine 386:
  [failed to restore the stack]

Goroutine 217 (running) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:501 +0x61b
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107

Goroutine 386 (finished) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:501 +0x61b
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107
==================
==================
WARNING: DATA RACE
Read at 0x00c42081b3a0 by goroutine 107:
  runtime.slicecopy()
      /home/ibk/go/src/runtime/slice.go:160 +0x0
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Get()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:118 +0x1d8
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Merge()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:59 +0x76
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func2()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:497 +0x37e

Previous write at 0x00c42081b3a0 by goroutine 241:
  runtime.slicecopy()
      /home/ibk/go/src/runtime/slice.go:160 +0x0
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:88 +0x175
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:54 +0x1e2
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Merge()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:82 +0x1b3
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func2()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:497 +0x37e

Goroutine 107 (running) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:501 +0x61b
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107

Goroutine 241 (finished) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:501 +0x61b
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107
==================
==================
WARNING: DATA RACE
Read at 0x00c42081baa0 by goroutine 303:
  runtime.slicecopy()
      /home/ibk/go/src/runtime/slice.go:160 +0x0
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Get()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:118 +0x1d8
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Merge()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:59 +0x76
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func2()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:497 +0x37e

Previous write at 0x00c42081baa0 by goroutine 377:
  ??()
      -:0 +0xffffffffffffffff
  net.(*pollDesc).prepareRead()
      /home/ibk/go/src/net/fd_poll_runtime.go:67 +0x41
  net.(*netFD).Read()
      /home/ibk/go/src/net/fd_unix.go:239 +0xd3
  net.(*conn).Read()
      /home/ibk/go/src/net/net.go:181 +0x96
  net.(*TCPConn).Read()
      <autogenerated>:72 +0x72
  bufio.(*Reader).fill()
      /home/ibk/go/src/bufio/bufio.go:97 +0x1a1
  bufio.(*Reader).ReadSlice()
      /home/ibk/go/src/bufio/bufio.go:338 +0x1c4
  github.com/zero-os/0-Disk/vendor/github.com/garyburd/redigo/redis.(*conn).readLine()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/vendor/github.com/garyburd/redigo/redis/conn.go:389 +0x6e
  github.com/zero-os/0-Disk/vendor/github.com/garyburd/redigo/redis.(*conn).readReply()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/vendor/github.com/garyburd/redigo/redis/conn.go:462 +0x53
  github.com/zero-os/0-Disk/vendor/github.com/garyburd/redigo/redis.(*conn).Do()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/vendor/github.com/garyburd/redigo/redis/conn.go:610 +0x1e4
  github.com/zero-os/0-Disk/vendor/github.com/garyburd/redigo/redis.(*pooledConnection).Do()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/vendor/github.com/garyburd/redigo/redis/pool.go:400 +0x12d
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).getContent.func1()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:137 +0x220
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).getContent()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:139 +0xbf
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Get()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:90 +0x1d6
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func2()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:488 +0xad

Goroutine 303 (running) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:501 +0x61b
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107

Goroutine 377 (finished) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:501 +0x61b
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107
==================
==================
WARNING: DATA RACE
Write at 0x00c420e35d38 by goroutine 389:
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:88 +0x187
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Delete()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:98 +0x61
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Delete()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:106 +0x98
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func3()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:518 +0x96

Previous write at 0x00c420e35d38 by goroutine 528:
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:88 +0x187
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Delete()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:98 +0x61
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Delete()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:106 +0x98
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func3()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:518 +0x96

Goroutine 389 (running) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:534 +0x7b6
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107

Goroutine 528 (running) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:534 +0x7b6
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107
==================
==================
WARNING: DATA RACE
Write at 0x00c420e35d38 by goroutine 310:
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:88 +0x187
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Delete()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:98 +0x61
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Delete()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:106 +0x98
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func3()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:518 +0x96

Previous write at 0x00c420e35d38 by goroutine 528:
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:88 +0x187
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Delete()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:98 +0x61
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Delete()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:106 +0x98
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func3()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:518 +0x96

Goroutine 310 (running) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:534 +0x7b6
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107

Goroutine 528 (running) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:534 +0x7b6
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107
==================
==================
WARNING: DATA RACE
Write at 0x00c420e35d38 by goroutine 264:
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:88 +0x187
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Delete()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:98 +0x61
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Delete()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:106 +0x98
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func3()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:518 +0x96

Previous write at 0x00c420e35d38 by goroutine 528:
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Set()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:88 +0x187
  github.com/zero-os/0-Disk/nbdserver/lba.(*LBA).Delete()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/lba/lba.go:98 +0x61
  github.com/zero-os/0-Disk/nbdserver/ardb.(*dedupedStorage).Delete()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped.go:106 +0x98
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock.func3()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:518 +0x96

Goroutine 264 (running) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:534 +0x7b6
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107

Goroutine 528 (running) created at:
  github.com/zero-os/0-Disk/nbdserver/ardb.testBackendStorageDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/ardb_test.go:534 +0x7b6
  github.com/zero-os/0-Disk/nbdserver/ardb.TestDedupedDeadlock()
      /home/ibk/pathgo/src/github.com/zero-os/0-Disk/nbdserver/ardb/deduped_test.go:175 +0x125
  testing.tRunner()
      /home/ibk/go/src/testing/testing.go:657 +0x107
==================
--- FAIL: TestDedupedDeadlock (3.50s)
	testing.go:610: race detected during execution of test
FAIL
exit status 1
FAIL	github.com/zero-os/0-Disk/nbdserver/ardb	3.552s
```

**Comment by iwanbk** [comment]
```
go test -race -run ^TestGetDedupedRoot
```
was not broken

File: nbdserver/lba/lba.go [hunk 1]
```diff
@@ -60,6 +60,7 @@ type LBA struct {
 // When a key is updated, the shard containing this blockindex is marked as dirty and will be
 // stored in the external metadataserver when Flush is called.
 func (lba *LBA) Set(blockIndex int64, h zerodisk.Hash) (err error) {
+	shardIndex := blockIndex / NumberOfRecordsPerLBAShard
```
Comment (GlenDC) [inline nbdserver/lba/lba.go]: Why this change? This doesn't change anything, right?
